ports/rp2/mbedtls/mbedtls_config.h

```diff
@@ -94,6 +94,8 @@
 #define MBEDTLS_X509_CRT_PARSE_C
 #define MBEDTLS_X509_USE_C
 
+#define MBEDTLS_BASE64_C
+#define MBEDTLS_PEM_PARSE_C
```

Comment:
please split this change into the rp2 commit (so that there's on commit that just touches `extmod/modussl_mbedtls.c`)

Reply:
Done!

Reply:
Thanks.

But another point of discussion is whether enabling PEM parsing is a good idea.  If it's enabled here then it sets a precedence that other ports should follow, otherwise writing portable code is hard (eg using a PEM in code on rp2 won't then work on an stm32 board).

PEM parsing adds a lot of unnecessary overhead (code size and RAM and time), it's much simpler to use DER format.  PEM is essentially base-64 encoded DER with header/footer data.  So to parse PEM means allocating a chunk of memory to hold the DER and then base-64 decoding the PEM into that buffer, and then using the DER (eg parsing it).

I think for MicroPython we should standardise on using (and documenting how to use) DER format only.  I'd be interested to hear your thoughts on this.

In order not to hold up this PR any longer, would be good not enable PEM here, and instead discuss it more widely.

Reply:
@daviesian any thoughts on this point?  My suggestion is to not add these 2 lines in this PR.

Reply:
@daviesian ping

Reply:
Sorry for missing this! You're quite right that this doesn't really belong here, although I do think I would be in favour of including it by default. Removed!

Reply:
FYI @dpgeorge if cadata is in DER format it can contain only one certificate, so in other words without PEM parsing enabled (and PEM data) this function can't load a cert chain (probably should use `mbedtls_x509_crt_parse_der` instead and load one DER at a time).
```
 *                 For certificates in PEM encoding, this may be a concatenation
 *                 of multiple certificates; for DER encoding, the buffer must
 *                 comprise exactly one certificate.
 * int mbedtls_x509_crt_parse( mbedtls_x509_crt *chain, const unsigned char *buf, size_t buflen );
```